lib/publiccloud/azure.pm

```diff
@@ -67,8 +68,29 @@ sub find_img {
     my $sku = (check_var('PUBLIC_CLOUD_AZURE_SKU', 'gen2')) ? 'V2' : 'V1';
     $name =~ s/\.xz$//;
     $name =~ s/\.vhdfixed$/-$sku.vhd/;
+
+    my $container = $self->container;
+    my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
+    my $key = $self->get_storage_account_keys($storage_account);
+    my $cmd_show = "az storage blob show --account-key $key -o json " .
+      "--container-name '" . $container . "' --account-name '$storage_account' --name $name " .
+      '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"';
+    record_info('BLOB INFO', script_output($cmd_show));    # This is temporal debug
+    my $start_time = time();
+    my $upload_time = 600;
+    my $completed = 0;
+    while (time() - $start_time < $upload_time) {
+        my $blob_info = decode_azure_json(script_output($cmd_show));
+        record_info('MD5', "Blob MD5: $blob_info->{md5}");
+        # Check that MD5 metadata exists and is not empty
+        # We don't compare the value yet, because we don't always have the downloaded.
+        $completed = 1 if ($blob_info->{md5} && length($blob_info->{md5}) > 0);
```

Comment:
```
length($blob_info->{md5}) > 0
```

formally it is never 0, even if blob does not have the MD5, the `az storage blob show` return you the string `null` or something similar.

Reply:
Pity that you did not raise this concern yesterday.

I cannot compare it with the md5 hash of the file. See the comment above.

The md5 hash should have always 32 characters, so please check now.